cluster/addons/dns/kube2sky/kube2sky.go

```diff
@@ -128,22 +131,40 @@ func newEtcdClient() (client *etcd.Client) {
 
 // TODO: evaluate using pkg/client/clientcmd
 func newKubeClient() (*kclient.Client, error) {
-	config := &kclient.Config{}
-
-	masterHost := os.Getenv("KUBERNETES_RO_SERVICE_HOST")
-	if masterHost == "" {
-		log.Fatalf("KUBERNETES_RO_SERVICE_HOST is not defined")
-	}
-	masterPort := os.Getenv("KUBERNETES_RO_SERVICE_PORT")
-	if masterPort == "" {
-		log.Fatalf("KUBERNETES_RO_SERVICE_PORT is not defined")
+	var config *kclient.Config
+	if *kubecfg_file == "" {
+		// No kubecfg file provided. Use kubernetes_ro service.
+		masterHost := os.Getenv("KUBERNETES_RO_SERVICE_HOST")
+		if masterHost == "" {
+			log.Fatalf("KUBERNETES_RO_SERVICE_HOST is not defined")
```

Comment:
(I think someone else was planning to overhaul this code, so we can wait for that to fix it.)